lib/rpmi_service_group_syssusp.c

```diff
@@ -172,19 +172,19 @@ static enum rpmi_error rpmi_syssusp_process_events(struct rpmi_service_group *gr
 
 	switch (sgsusp->current_state) {
 	case RPMI_SYSSUSP_STATE_SUSPEND_PENDING:
-		if (!sgsusp->ops->system_suspend_ready(sgsusp->ops_priv,
-						       sgsusp->current_hart_index))
-			return RPMI_SUCCESS;
+		if (sgsusp->ops->system_suspend_ready(sgsusp->ops_priv,
+						      sgsusp->current_hart_index))
+			return RPMI_ERR_BUSY;
 		sgsusp->ops->system_suspend_finalize(sgsusp->ops_priv,
 						sgsusp->current_hart_index,
 						sgsusp->current_syssusp_type,
 						sgsusp->current_resume_addr);
 		sgsusp->current_state = RPMI_SYSSUSP_STATE_SUSPENDED;
 		break;
 	case RPMI_SYSSUSP_STATE_SUSPENDED:
-		if (!sgsusp->ops->system_suspend_can_resume(sgsusp->ops_priv,
-							    sgsusp->current_hart_index))
-			return RPMI_SUCCESS;
+		if (sgsusp->ops->system_suspend_can_resume(sgsusp->ops_priv,
+							   sgsusp->current_hart_index))
+			return RPMI_ERR_BUSY;
```

Comment:
IMO there are two things missing in librpmi handling of such case -
- A error case like `RPMI_ERR_EAGAIN` which is more appropriate for such scenario.  We don't have such in the RPMI message protocol (not yet needed) but we can have an internal error code in librpmi.
- Handling of such error case in the rpmi context which does not means the failure but a hint to the caller that it has to be tried again.